lib/text/ui_text_displayer.js

```diff
@@ -996,15 +979,17 @@ shaka.text.UITextDisplayer = class {
 
     const {unit, value} = lengthValueInfo;
 
+    const realValue = value * scaleFactor;
+
     switch (unit) {
       case '%':
         return shaka.text.UITextDisplayer.getAbsoluteLengthInPixels_(
-            value / 100, cue, videoContainer);
+            realValue / 100, cue, videoContainer);
       case 'c':
         return shaka.text.UITextDisplayer.getAbsoluteLengthInPixels_(
-            value, cue, videoContainer);
+            realValue, cue, videoContainer);
       default:
-        return lengthValue;
+        return String(realValue + unit);
```

Comment:
I don't think you need this string cast. `unit` is a string, and adding a number and a string already results in a string in Javascript.